src/overlays/actors/ovl_En_Bombf/z_en_bombf.c

```diff
@@ -21,25 +32,471 @@ const ActorInit En_Bombf_InitVars = {
+    }
+}
+
+void EnBombf_Explode(EnBombf* this, GlobalContext* globalCtx) {
+    Player* player;
+
+    if (this->jntSphList.list->dim.modelSphere.radius == 0) {
+        this->actor.flags |= 0x20;
+        func_800AA000(this->actor.xzDistFromLink, 0xFF, 0x14, 0x96);
+    }
+
+    this->jntSphList.list->dim.modelSphere.radius += 8;
+    this->jntSphList.list->dim.worldSphere.radius = this->jntSphList.list->dim.modelSphere.radius;
+
+    if (this->actor.params == 1) {
+        CollisionCheck_SetAT(globalCtx, &globalCtx->colChkCtx, &this->jntSphList.base);
+    }
+
+    if (globalCtx->envCtx.unk_8C[3] != 0) {
+        globalCtx->envCtx.unk_8C[3] -= 0x19;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Bombf/EnBombf_Destroy.s")
+    if (globalCtx->envCtx.unk_8C[4] != 0) {
+        globalCtx->envCtx.unk_8C[4] -= 0x19;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Bombf/func_809C7180.s")
+    if (globalCtx->envCtx.unk_8C[5] != 0) {
+        globalCtx->envCtx.unk_8C[5] -= 0x19;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Bombf/func_809C71A8.s")
+    if (globalCtx->envCtx.unk_8C[0] != 0) {
+        globalCtx->envCtx.unk_8C[0] -= 0x19;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Bombf/func_809C74AC.s")
+    if (globalCtx->envCtx.unk_8C[1] != 0) {
+        globalCtx->envCtx.unk_8C[1] -= 0x19;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Bombf/func_809C75C8.s")
+    if (globalCtx->envCtx.unk_8C[2] != 0) {
+        globalCtx->envCtx.unk_8C[2] -= 0x19;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Bombf/func_809C7624.s")
+    if (this->timer == 0) {
+        player = PLAYER;
 
+        if ((player->stateFlags1 & 0x800) && (player->heldActor == &this->actor)) {
+            player->actor.attachedB = NULL;
+            player->heldActor = NULL;
+            player->interactRangeActor = NULL;
+            player->stateFlags1 &= ~0x800;
+        }
+
+        Actor_Kill(&this->actor);
+    }
+}
+
+#ifdef NON_MATCHING
+// stack issues. exact problems are commented in the function
+void EnBombf_Update(Actor* thisx, GlobalContext* globalCtx) {
+    Vec3f effVelocity = { 0.0f, 0.0f, 0.0f };
+    Vec3f bomb2Accel = { 0.0f, 0.1f, 0.0f };
+    Vec3f effAccel = { 0.0f, 0.0f, 0.0f };
+    Vec3f effPos;
+    Vec3f dustAccel = { 0.0f, 0.6f, 0.0f };
+    Color_RGBA8_n dustColor = { 255, 255, 255, 255 };
+    EnBombf* this = THIS;
+    s32 pad1;
+    s32 pad2;
+
+    if ((this->unk_200 != 0) && (this->timer != 0)) {
+        this->timer--;
+    }
+
+    // activate bump collider if link is 20 or more units away from the bomb in the xz plane
+    // or if the height difference between link and the bomb is 80 or more units
+    if ((!this->bumpOn) && (!func_8002F410(&this->actor, globalCtx)) &&
+        ((this->actor.xzDistFromLink >= 20.0f) || (ABS(this->actor.yDistFromLink) >= 80.0f))) {
+        this->bumpOn = true;
+    }
```

Comment:
Same here about the usefulness of these comments like I mentioned for en_bom. Tho this version of the check also includes a function call so that might be worth explaining briefly on its own.

Reply:
Also, regarding cleanliness:
```suggestion
    // activate bump collider if link is 20 or more units away from the bomb in the xz plane
    // or if the height difference between link and the bomb is 80 or more units
    if (!this->bumpOn && !func_8002F410(&this->actor, globalCtx) &&
        ((this->actor.xzDistFromLink >= 20.0f) || (ABS(this->actor.yDistFromLink) >= 80.0f))) {
        this->bumpOn = true;
    }
```

Reply:
its just an attachedA check. hoping we can name these attached null check functions soon cause theyre used so much and make reading the code much easier